packages/drivers/routerlicious-driver/src/test/r11sSocketTests.spec.ts

```diff
@@ -69,129 +99,63 @@ describe("R11s Socket Tests", () => {
 		)) as DocumentService;
 	});
 
-	it("connect_document_error with Token Revoked error", async () => {
-		const errorToThrow = {
-			code: 403,
-			message: "TokenRevokedError",
-			retryAfterMs: 10,
-			internalErrorCode: "TokenRevoked",
-			errorType: DriverErrorTypes.authorizationError,
-			canRetry: false,
-		};
-		const errorEventName = "connect_document_error";
-		socket = new ClientSocketMock({
-			connect_document: { eventToEmit: errorEventName, errorToThrow },
-		});
-
-		await assert.rejects(
-			mockSocket(socket as unknown as Socket, async () =>
-				documentService.connectToDeltaStream(client),
-			),
-			{
-				errorType: DriverErrorTypes.authorizationError,
-				scenarioName: "connect_document_error",
-				internalErrorCode: "TokenRevoked",
-			},
-			"Error should have occurred",
-		);
-	});
-
-	it("Socket error with Token Revoked error", async () => {
-		const errorToThrow = {
-			code: 403,
-			message: "TokenRevokedError",
-			retryAfterMs: 10,
-			internalErrorCode: "TokenRevoked",
-			errorType: DriverErrorTypes.authorizationError,
-			canRetry: false,
-		};
-		const errorEventName = "connect_document_success";
-		socket = new ClientSocketMock({
-			connect_document: { eventToEmit: errorEventName },
-		});
-
-		const connection = await mockSocket(socket as unknown as Socket, async () =>
-			documentService.connectToDeltaStream(client),
-		);
-		let error: IAnyDriverError | undefined;
-		connection.on("disconnect", (reason: IAnyDriverError) => {
-			error = reason;
-		});
-
-		// Send Token Revoked error
-		socket.sendErrorEvent(errorToThrow);
-
-		assert(
-			error?.errorType === DriverErrorTypes.authorizationError,
-			"Error type should be authorizationError",
-		);
-		assert(error.scenarioName === "error", "Error scenario name should be error");
-		assert(
-			(error as any).internalErrorCode === "TokenRevoked",
-			"Error internal code should be TokenRevoked",
-		);
-	});
-
-	it("connect_document_error with Cluster Draining error", async () => {
-		const errorToThrow = {
-			code: 503,
-			message: "ClusterDrainingError",
-			retryAfterMs: 1000,
-			internalErrorCode: R11sServiceClusterDrainingErrorCode,
-			errorType: RouterliciousErrorTypes.clusterDrainingError,
-			canRetry: true,
-		};
-		const errorEventName = "connect_document_error";
-		socket = new ClientSocketMock({
-			connect_document: { eventToEmit: errorEventName, errorToThrow },
+	describe("on 'connect_document_error'", () => {
+		errorScenarios.forEach((scenario) => {
+			it(`should reject the connection promise for a ${scenario.name} error`, async () => {
+				const socket = new ClientSocketMock({
+					connect_document: {
+						eventToEmit: "connect_document_error",
+						errorToThrow: scenario.errorToThrow,
+					},
+				});
+
+				await assert.rejects(
+					mockSocket(socket as unknown as Socket, async () =>
```

Comment:
Is this cast through `unknown` necessary? Does `ClientSocketMock` not satisfy `Socket`?

If not, would it make more sense to update `mockSocket` to take `ClientSocketMock` directly? Then at least we could reduce the number of places we have to cast through `unknown` (and we can document why 😉)

Reply:
@Josmithr  I updated `mockSocket` to accept `ClientSocketMock`, which removed the casts at call. But, we still need one cast inside the function because:

1. Sinon's `returns()` is strongly typed and expects a `Socket`
2. `ClientSocketMock` only implements the subset needed for tests (missing 50+ `Socket` properties)

Added a comment